### PR TITLE
Move Policy CRUD cmdlets to autorest generation: fixes (#23518)

### PR DESCRIPTION
**Pull request by mentat9**
* Set all tests to -LiveOnly Add back some test infrastructure fixes that were lost Improve parameter handling for Remove-AzPolicySetDefinition

* Move to supported way of sharing Helpers.psm1 file

* Switch to recommended approach for shared functions

## Mandatory Checklist

- Please choose the target release of Azure PowerShell
  - [X] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] N/A

- [X] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Resources/Policy.Autorest/custom/Remove-AzPolicySetDefinition.ps1 [hunk 1]
```diff
@@ -35,8 +35,8 @@ function Remove-AzPolicySetDefinition {
 [CmdletBinding(DefaultParameterSetName='Name', SupportsShouldProcess, ConfirmImpact='High')]
 param(
     [Parameter(ParameterSetName='Name', Mandatory, ValueFromPipelineByPropertyName)]
-    [Parameter(ParameterSetName='ManagementGroupName', ValueFromPipelineByPropertyName)]
-    [Parameter(ParameterSetName='SubscriptionId', ValueFromPipelineByPropertyName)]
+    [Parameter(ParameterSetName='ManagementGroupName', Mandatory)]
```
Comment (msJinLei) [inline src/Resources/Policy.Autorest/custom/Remove-AzPolicySetDefinition.ps1:38]: It is a breaking change to change from optional to mandatory

**Review by msJinLei — CHANGES_REQUESTED** [review]
Please resolve the conflicts

**Review by VeryEarly — APPROVED** [review]

**Pull request merged**